decentraland/unity-renderer
# fix: memory leak caused by cancellations

## What does this PR change?

Fixes a Memory Leak caused by textures being created asynchronously and the process being canceled before those textures were handled correctly for their deletion.
This was introduced with #1761 

## How to test the changes?

Unity Only:
- Open up APK_GLTF_InteractiveTest scene and run it
- Open up the Profiler and check out current Texture and Mesh memory and **save those numbers**.
- Press Z to spawn GLTFs and X to delete the last one loaded, you can press C for an automated and randomized Loading and Deletion and press again to toggle it Off.
- Press X until no GLTFs are loaded
- Check the Texture and Memory usage and compare it with the initial values, they must be equal or somewhat close (< 1 MB of tolerance )

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
